commit 9003c4e220c2954a53c5da0d739ed15a46c13429
Author: Waldemar Rymarkiewicz <waldemar.rymarkiewicz@tieto.com>
Date:   Thu Apr 28 12:07:54 2011 +0200

    Bluetooth: Don't modify sec_level if auth failed
    
    If authentication fails the security level should stay as it was set
    before the process has started. Setting BT_SECURITY_LOW can hide real
    security level on a link eg. having BT_SECURITY_MEDIUM on the link,
    re-authenticate with failure to get BT_SECURITY_HIGH, as  a result we
    get BT_SECURITY_LOW on the link while the real security is still medium.
    
    Signed-off-by: Waldemar Rymarkiewicz <waldemar.rymarkiewicz@tieto.com>
    Signed-off-by: Gustavo F. Padovan <padovan@profusion.mobi>

diff --git a/net/bluetooth/hci_event.c b/net/bluetooth/hci_event.c
index fbbb63f8a897..35f98980070c 100644
--- a/net/bluetooth/hci_event.c
+++ b/net/bluetooth/hci_event.c
@@ -1459,7 +1459,6 @@ static inline void hci_auth_complete_evt(struct hci_dev *hdev, struct sk_buff *s
 			conn->sec_level = conn->pending_sec_level;
 		} else {
 			mgmt_auth_failed(hdev->id, &conn->dst, ev->status);
-			conn->sec_level = BT_SECURITY_LOW;
 		}
 
 		clear_bit(HCI_CONN_AUTH_PEND, &conn->pend);